### PR TITLE
feature: S3C-3584 assume role backbeat client

### PR DESCRIPTION
**Pull request by saitharunthodupunuru**
**What is Changed?**
* Restructured the unit tests by service
* Create Marshaller for AssumeRoleBackbeat request
* Create Generic Unmarshaller for AWS complaint responses
* Implement AssumeRoleBackbeat client
* Unit tests for AssumeRoleBackbeat client

### CODE REVIEW

**Comment by RachedBenMustapha** [comment]
Is this really something we need to expose through the Java client? It's an internal API used by backbeat which isn't written in Java.

**Comment by saitharunthodupunuru** [comment]
> Is this really something we need to expose through the Java client? It's an internal API used by backbeat which isn't written in Java.

According to the planned design in PR https://github.com/scality/vmware-ose-scality/pull/23, we are going to use `AssumeRoleBackbeat` API from OSIS to assume the role of the account to perform IAM operations on the account.

**Review by anurag4DSB — APPROVED** [review]

File: vaultclient/src/main/java/com/scality/vaultclient/dto/GenericAWSResponseJsonUnmarshaller.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+                .setFieldNamingPolicy(FieldNamingPolicy.UPPER_CAMEL_CASE)
+                .create();
+                result = gson.fromJson(response, typeParameterClass);
+            }
+        } catch (Exception e) {
+            throw new VaultClientException("Unable to unmarshall response content", e);
+        }
+        return result;
+    }
+}
```
Comment (anurag4DSB) [inline vaultclient/src/main/java/com/scality/vaultclient/dto/GenericAWSResponseJsonUnmarshaller.java:49]: missing new line

Comment (saitharunthodupunuru) [reply vaultclient/src/main/java/com/scality/vaultclient/dto/GenericAWSResponseJsonUnmarshaller.java]: done in fc327f5df2c3c7beca29f69d2410f76cc8de7b1b

**Review by ilkescality — APPROVED** [review]

**Pull request merged**